use-cases/xtclim/hpo.py

```diff
@@ -0,0 +1,259 @@
+import argparse
+import os
+import sys
+from pathlib import Path
+from typing import Dict
+import yaml
+
+import matplotlib.pyplot as plt
+import ray
+import torch
+from ray import train, tune
+
+from itwinai.parser import ConfigParser
+
+sys.path.append(os.path.join(os.path.dirname(__file__), 'src'))
+sys.path.append(os.path.join(os.path.dirname(__file__), 'preprocessing'))
+
+
+def run_trial(config: Dict, data: Dict):
+    """Execute a single trial using the given configuration (config).
+    This runs a full training pipeline - you can also specify a pipeline as a dictionary,
+    e.g. if you only want to run certain parts without changing your config.yaml file
+    (see below).
+
+    Args:
+        config (dict): A dictionary containing hyperparameters, such as:
+            - 'batch_size' (int): The size of the batch for training.
+            - 'lr' (float): The learning rate for the optimizer.
+        data (dict): A dictionary containing a "pipeline_path" field, which points to the yaml
+            file containing the pipeline definition
+    """
+    with open('pipeline.yaml', 'r') as f:
+        yaml_config = yaml.safe_load(f)
+
+    # Override keys of hyperparameters to be tuned from Ray config
+    yaml_config['batch_size'] = config['batch_size']
+    yaml_config['lr'] = config['lr']
+
+    # Set loggers to None as Ray logs the runs
+    yaml_config['pipeline']['init_args']['steps']['training-step']['init_args']['logger'] = None
+    yaml_config['pipeline']['init_args']['steps']['evaluation-step']['init_args']['logger'] = None
```

Comment:
These can also be overridden using the override_keys in the ConfigParser: https://itwinai.readthedocs.io/latest/python-api/itwinai.parser.html#itwinai.parser.ConfigParser
(see comment below)